ensure older grpc versions remain guarded

javacruft opened the pull request:
grpc <= 1.70 requires protobuf-dev=3.29.5; however this version of the
package is old/unguarded and not installable any longer.

Introduce a new protobuf-25.9 (using versioning aligned to current
protobuf package) along versioned subpackages, using the SOVERSION for
all of the shared library packages.

protobuf-25.9 and grpc <= 1.70 also need an older version of abseil-cpp.

Introduce a new abseil-cpp-20250127 package along with versioned subpackages
using the SOVERSION for the shared library package ensuring co-installabilty
with newer versions.

Update grpc <= 1.70, falco and libpulsar to use new versioned packages.

javacruft committed:
protobuf-25.9: fix issues with older grpc's

grpc <= 1.70 requires protobuf-dev=3.29.5; however this version of the
package is old/unguarded and not installable any longer.

Introduce a new protobuf-25.9 (using versioning aligned to current
protobuf package) along versioned subpackages, using the SOVERSION for
all of the shared library packages.

Update grpc <= 1.70 to use new package.

javacruft committed:
Align build configuration with older version.

javacruft committed:
Sync more changes from older package version.

javacruft committed:
grpc-1.67: disable GCC hardening

Brings this version back in line with other grpc version streams and
resolves the current FTBFS.

sil2100 reviewed (commented):
This looks good, just trying to understand the rationale for using <major.minor.0> as the library naming scheme. In general we're rather only keying on the major revision, right? At most major.minor should be enough? Can you tell me more?

javacruft commented:
> This looks good, just trying to understand the rationale for using <major.minor.0> as the library naming scheme. In general we're rather only keying on the major revision, right? At most major.minor should be enough? Can you tell me more?

It all rather depends on the project being packaged and the soversion used in the projects build:

https://apk.chaindag.dev/https/apk.cgr.dev/wolfi-presubmit/e551a581757af39b50d92ff609f68e57271b5bce/aarch64/libprotobuf-lite29.5.0-29.5-r0.apk@etag:4d12f38e89dd71110406cf2f1c715b28/.PKGINFO

I *think* I have this right for protobuf - but ultimately so long as we generate a unique package name we're OK

javacruft committed:
Merge branch 'main' into grpc-protobuf/version-stream

javacruft committed:
grpc-*: rebump post merge of main.

javacruft committed:
grpc-1.{66-70}: ensure guarded

Add new abseil-cpp-20250127 versioned package to support other packages
which are not compatible with newer abseil releases.

Update existing abseil-cpp-dev version pins to use new versioned
package.

Drop pins for re2 - this is no longer required as the new abseil-cpp
package ships a co-installable libabsl package.

javacruft committed:
bump epochs

javacruft committed:
abseil-cpp-20250127: fix tests

javacruft committed:
protobuf-29.5: use versioned abseil-cpp for compat

javacruft committed:
falco: fix protobuf dev package version in tests

javacruft committed:
falco: ensure correct -dev package dependencies are used.

justinvreeland reviewed (dismissed):
Just a small change to remove the duplicate re2-dev lines. We don't need to include re2 since it should(?) resolve correctly without the conflicts on abseil.

javacruft committed:
Fix duplicated re2-dev dependency.

sil2100 commented on grpc-1.66.yaml:
```diff
@@ -140,12 +136,11 @@ subpackages:
-        - re2=2025.07.17-r1
-        - re2-dev=2025.07.17-r1
+        - re2-dev
+        - re2-dev
```
Duplicate, remove please!

sil2100 commented on grpc-1.67.yaml:
```diff
@@ -156,12 +152,11 @@ subpackages:
-        - re2=2025.07.17-r1
-        - re2-dev=2025.07.17-r1
+        - re2-dev
+        - re2-dev
```
Same here ^

sil2100 commented on grpc-1.69.yaml:
```diff
@@ -156,12 +152,11 @@ subpackages:
-        - re2=2025.07.17-r1
-        - re2-dev=2025.07.17-r1
+        - re2-dev
+        - re2-dev
```
Same here ^

sil2100 reviewed (changes requested):
This looks good but there's still some `re2-dev` duplicated lines. Once those are cleaned up, we're good to merge.

javacruft committed:
Drop more duplicated re2-dev

sil2100 reviewed (approved):
LGTM now!
